src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -985,11 +1021,37 @@ namespace OpenLoco::Ui::Windows::PromptBrowse
     }
 
     // 0x00446E87
-    // TODO: only called by this window -- implement.
-    static void sub_446E87(Window* self)
+    static void loadFileHeader(Window* self)
     {
-        registers regs;
-        regs.esi = X86Pointer(self);
-        call(0x00446E87, regs);
+        freeFileHeader();
+
+        _9DA285 = 0;
+        if (self->var_85A == -1)
+            return;
+
+        file_entry& entry = _files[self->var_85A];
```

Comment:
Whilst we are in this function can you apply the coding style. Make this `FileEntry` and make all the functions camelCase.